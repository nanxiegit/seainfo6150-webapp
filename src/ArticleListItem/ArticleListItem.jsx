import React from "react";

const ArticleListItem = ({props}) => {
    return (
        <article>
            <h2>{props.title}</h2>
            <time dateTime={props.timeStamp}>{props.displayDate}</time>
            <p>{props.shortText}</p>
        </article>

    );
};


export default ArticleListItem;