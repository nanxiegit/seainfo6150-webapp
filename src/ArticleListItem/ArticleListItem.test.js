import React from "react";
import ArticleListItem from "./ArticleListItem";

describe("ArticleListItem tests", () => {
  it("renders correctly", () => {
    const article = {
        "image": {
            "_url": "https://cdn.cnn.com/cnnnext/dam/assets/170321135754-airport-security-line-file-super-169.jpg"
          }
    };
    const { container } = render(<ArticleListItem article={article} />);
    expect(container).toMatchSnapshot();
  });
});