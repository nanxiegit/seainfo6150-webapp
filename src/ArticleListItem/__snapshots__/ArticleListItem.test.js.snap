// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleListItem tests renders correctly 1`] = `
<div>
  <li
    class="container"
  >
    <article
      class="article"
    >
      <header
        class="header"
      >
        <img
          class="image"
          src="https://cdn.cnn.com/cnnnext/dam/assets/170321135754-airport-security-line-file-super-169.jpg"
        />
        <h2
          class="title"
        >
          <a
            class="link"
            href="/articlelist/undefined"
          />
        </h2>
      </header>
      <div
        class="wrapper"
      />
      <button
        class="button"
      >
        Show more
      </button>
    </article>
  </li>
</div>
`;
